src/_components/shared/page_navigation.rb

```diff
@@ -0,0 +1,27 @@
+class Shared::PageNavigation < Bridgetown::Component
+  def initialize(entries:, current:)
+    @entries = entries
+    @current = current
+
+    set_links
+  end
+
+  def set_links
+    pages = []
+    get_resources(@entries, pages)
```

Comment:
I would create a logic to set
```suggestion
     pages = get_pages(@entries)
```